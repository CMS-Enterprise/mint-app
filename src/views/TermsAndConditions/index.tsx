import React from 'react';
import { useTranslation } from 'react-i18next';

import MainContent from 'components/MainContent';
import PageHeading from 'components/PageHeading';

const TermsAndConditions = () => {
  const { t } = useTranslation('termsAndConditions');
  return (
    <MainContent className="grid-container line-height-body-5">
      <PageHeading>{t('heading')}</PageHeading>
      <h2>{t('subheading')}</h2>
      <p className="margin-y-4">{t('unauthorizedUse')}</p>
      <p className="margin-y-4">{t('socialMediaUse')}</p>
      <p className="margin-y-4">{t('consent')}</p>
      <p className="margin-y-4">{t('infoStorage')}</p>
    </MainContent>
  );
};

export default TermsAndConditions;
