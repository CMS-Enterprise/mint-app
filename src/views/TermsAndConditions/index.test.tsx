import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

import TermsAndConditions from './index';

vi.mock('@okta/okta-react', () => ({
  useOktaAuth: () => {
    return {
      authState: {
        isAuthenticated: true
      },
      oktaAuth: {
        getUser: async () => {},
        logout: async () => {}
      }
    };
  }
}));

describe('The Terms & Conditions page', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <TermsAndConditions />
      </MemoryRouter>
    );
    expect(screen.getByText('Terms & Conditions')).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <TermsAndConditions />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
