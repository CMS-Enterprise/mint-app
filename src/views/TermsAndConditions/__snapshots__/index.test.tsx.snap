// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Terms & Conditions page matches the snapshot 1`] = `
<main
  className="easi-main-content grid-container line-height-body-5"
  id="main-content"
  role="main"
  tabIndex={-1}
>
  <h1
    aria-live="polite"
    className="easi-h1"
    tabIndex={-1}
  >
    Terms & Conditions
  </h1>
  <h2>
    This system is provided for Government authorized use only.
  </h2>
  <p
    className="margin-y-4"
  >
    Unauthorized or improper use of this system is prohibited and may result in disciplinary action and/or civil and criminal penalties.
  </p>
  <p
    className="margin-y-4"
  >
    Personal use of social media and networking sites on this system is limited as to not interfere with official work duties and is subject to monitoring.
  </p>
  <p
    className="margin-y-4"
  >
    By using this system, you understand and consent to the following: The Government may monitor, record, and audit your system usage, including usage of personal devices and email systems for official reasonable expectation of privacy regarding any communication or data transiting or stored on this system. At any time, and for any lawful Government purpose, the government may monitor, intercept, and search and seize any communication or data transiting or stored on this system.
  </p>
  <p
    className="margin-y-4"
  >
    Any communication or data transiting or stored on this system may be disclosed or used for any lawful Government purpose.
  </p>
</main>
`;
